js/src/forum/components/UserBio.tsx

```diff
@@ -1,43 +1,35 @@
 import app from 'flarum/forum/app';
-import Component from 'flarum/common/Component';
+import Component, { ComponentAttrs } from 'flarum/common/Component';
 import LoadingIndicator from 'flarum/common/components/LoadingIndicator';
 import Button from 'flarum/common/components/Button';
 import classList from 'flarum/common/utils/classList';
 import extractText from 'flarum/common/utils/extractText';
 
-/**
- * The `UserBio` component displays a user's bio, optionally letting the user
- * edit it.
- */
-export default class UserBio extends Component {
-  oninit(vnode) {
-    super.oninit(vnode);
-    /**
-     * Whether the bio is currently being edited.
-     *
-     * @type {boolean}
-     */
-    this.editing = false;
+import type Mithril from 'mithril';
+import type User from 'flarum/common/models/User';
+import type { NestedStringArray } from '@askvortsov/rich-icu-message-formatter';
 
-    /**
-     * Whether the bio is currently being saved.
-     *
-     * @type {boolean}
-     */
-    this.loading = false;
+export interface UserBioAttrs extends ComponentAttrs {
+  user: User;
+  editable: boolean;
+}
 
-    /**
-     * The rows to show in the textarea by default when editing.
-     * This is set to 5 by default, but can be overridden by the `--bio-max-lines` CSS variable.
-     *
-     * @type {string}
-     */
-    this.textareaRows = '5';
+export default class UserBio extends Component<UserBioAttrs> {
+  editing: boolean = false;
```

Comment:
I'm not sure if having values moved from `oninit` to instance creation causes any issues or not... theoretically `oninit` should only be called once, and since these are primitives it shouldn't really matter as there are no side effects? Just some thoughts.